tls/ttls.c

```diff
@@ -1334,10 +1334,37 @@ ttls_send_alert(TlsCtx *tls, unsigned char lvl, unsigned char msg)
 	io->alert[0] = lvl;
 	io->alert[1] = msg;
 
-	if (msg == TTLS_ALERT_MSG_CLOSE_NOTIFY)
+	switch (msg) {
+	case TTLS_ALERT_MSG_CLOSE_NOTIFY:
```

Comment:
Do we really need to enumerate all alerts here? If alert is fatal, we close. If it's a warning, but close_notify, we also close. So I think it's enough to add `|| lvl == TTLS_ALERT_LEVEL_FATAL` to the `if`.

There is only one place in the code where we issue a warning-level alert that is not close_notify. It's `TTLS_ALERT_MSG_NO_RENEGOTIATION`.  I think we need a fatal level there since we treat that's as an error, and refuse to proceed by returning `TTLS_ERR_UNEXPECTED_MESSAGE`.

Reply:
I can see changes in the patches that are addressing this comment.